### PR TITLE
[CPU][TESTS] Separate custom single layer tests per architecture

### PR DESCRIPTION
**Pull request by EgorDuplensky**
### Details:
 - The idea is to have common test instances for all the architectures
and specific ones if necessary

The following structure is expected:

# CPU plugin functional tests

## Structure of the platform specific tests

Taken CPU plugin specific single layer tests as an example:

``` shell
single_layer_tests/
├── classes # test classes
│   ├── activation.cpp # test class with common parameters
│   ├── activation.h
│   ├── convolution.cpp
│   └── convolution.h
└── instances # test instances
    ├── arm # arch specific instances if any
    │   ├── acl # backend specific instances if any
    │   │   └── actication.cpp 
    │   └── onednn
    │       └── convolution.cpp
    ├── common # common instances across all the architecture
    │   ├── activation.cpp
    │   └── onednn
    │       └── convolution.cpp
    ├── _some_new_arch
    │   ├── activation.cpp
    │   └── convolution.cpp
    └── x64 # arch specific instances if any
        ├── activation.cpp # native instances for the arch (no backend involved)
        ├── onednn # backend specific instances if any
        │   └── convolution.cpp
        └── _some_new_backend
            └── convolution.cpp
```
### Tickets:
 -

### CODE REVIEW

**Comment by EgorDuplensky** [comment]
@ilya-lavrenov @dmitry-gorokhov Could you please take a look?

**Comment by dmitry-gorokhov** [comment]
@EgorDuplensky I would propose to create Readme file inside tests folder and put there strcture example you have in the PR description + some brief text description. That would help developers to understand the structure and general idea on how it should be extended.

File: src/plugins/intel_cpu/tests/functional/target_per_test.cmake [hunk 1]
```diff
@@ -45,6 +44,47 @@ function(create_target_per_test_for_directory TEST_DIR TARGET_PREFIX)
...
+  foreach(TEST_CLASS_FILE ${LIST_OF_TEST_CLASSES})
+    get_filename_component(TEST_CLASS ${TEST_CLASS_FILE} NAME_WE)
+    get_filename_component(TEST_CLASS_FILE_NAME ${TEST_CLASS_FILE} NAME)
+
+    # find all the source files with the name of a class file
+    if(X86_64)
+        file(GLOB_RECURSE LIST_OF_TEST_ARCH_INSTANCES ${TEST_DIR}/instances/x64/${TEST_CLASS_FILE_NAME})
+    else()
+        file(GLOB_RECURSE LIST_OF_TEST_ARCH_INSTANCES ${TEST_DIR}/instances/arm/${TEST_CLASS_FILE_NAME})
+    endif()
```
[the remark this answers is not in the document]
Comment (EgorDuplensky) [reply src/plugins/intel_cpu/tests/functional/target_per_test.cmake:60]: Not sure how to make it generic.

File: src/plugins/intel_cpu/tests/functional/test_utils/cpu_test_utils.cpp [hunk 2]
```diff
@@ -253,6 +253,11 @@ CPUTestsBase::CPUInfo CPUTestsBase::getCPUInfo() const {
     return makeCPUInfo(inFmts, outFmts, priority);
 }
 
+#if defined(OPENVINO_ARCH_ARM) || defined(OPENVINO_ARCH_ARM64)
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/tests/functional/test_utils/cpu_test_utils.cpp:256]: It looks like we should use OV_CPU_WITH_ACL definition instead

Comment (EgorDuplensky) [reply src/plugins/intel_cpu/tests/functional/test_utils/cpu_test_utils.cpp]: Updated

File: src/plugins/intel_cpu/tests/functional/target_per_test.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (EgorDuplensky) [reply src/plugins/intel_cpu/tests/functional/target_per_test.cmake:60]: Updated

File: src/plugins/intel_cpu/tests/functional/CMakeLists.txt [hunk 3]
```diff
@@ -19,18 +19,33 @@ else()
     set(EXCLUDED_SOURCE_PATHS ${CMAKE_CURRENT_SOURCE_DIR}/extension ${CMAKE_CURRENT_SOURCE_DIR}/shared_tests_instances/onnx)
 endif()
 
-if(NOT X86_64)
+if(X86_64)
+    list(APPEND EXCLUDED_SOURCE_PATHS
+    ${CMAKE_CURRENT_SOURCE_DIR}/single_layer_tests/instances/arm
+    ${CMAKE_CURRENT_SOURCE_DIR}/subgraph_tests/arm)
+else()
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/tests/functional/CMakeLists.txt:26]: @EgorDuplensky I beleive we also sohuld split the branche (one logic for ARM and another for WASM, RISC etc).

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**